egs/wsj/s5/steps/nnet3/multilingual/allocate_multilingual_examples.py

```diff
@@ -95,21 +97,43 @@ def get_args():
         'output-2'.""",
         epilog="Called by steps/nnet3/multilingual/combine_egs.sh")
 
-    parser.add_argument("--samples-per-iter", type=int, default=40000,
+    parser.add_argument("--samples-per-iter", type=int, default=None,
                         help="The target number of egs in each archive of egs, "
-                        "(prior to merging egs). ")
+                        "(prior to merging egs). If provided, will "
+                        "override --frames-per-iter and --frames-per-eg-list")
+    parser.add_argument("--frames-per-iter", type=int, default=400000,
+                        help="The target number of frames in each archive of "
+                        "egs")
+    parser.add_argument("--frames-per-eg-list", type=str, default=None,
+                        action=common_lib.NullstrToNoneAction,
+                        help="Number of frames per eg for each input language "
+                        "as a comma separated list. If not provided, "
+                        "this is assumed to be 1 in each language. "
+                        "Note: This affects the proportion of different "
```

Comment:
Amount of input data (=number of frames) is proportional to frames per eg since this script only sees how many egs are there.

Reply:
I guess it won't affect the proportion but it would affect the number of archives and size of each archive. But if the mechanism to set number of archives is changed to sum(num_archives) as you mentioned then it might not have this issue.